app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,360 @@
+    }
+
+    fn set_model(&self, model: &Model) {
+        match model {
+            Model::Text(content) => self.switch_to_text(content),
+            Model::Separator => self.switch_to_separator(),
+            Model::Ellipsis => self.switch_to_ellipsis(),
+        }
+    }
+
+    fn switch_to_text(&self, content: &str) {
+        self.text.set_content(content);
+        if self.state.get() != State::Text {
+            self.remove_current();
+            self.display_object.add_child(&self.text);
+            self.state.set(State::Text);
+        }
+    }
+
+    fn switch_to_separator(&self) {
+        if self.state.get() != State::Separator {
+            self.remove_current();
+            self.display_object.add_child(&self.separator);
+            self.state.set(State::Separator);
+        }
+    }
+
+    fn switch_to_ellipsis(&self) {
+        if self.state.get() != State::Ellipsis {
+            self.remove_current();
+            self.display_object.add_child(&self.ellipsis);
+            self.state.set(State::Ellipsis);
+        }
+    }
+
+    fn update_layout(&self, contour: Contour, text_size: text::Size, text_offset: f32) {
+        let size = contour.size;
+        self.text.set_position_xy(Vector2(text_offset - size.x / 2.0, text_size.raw / 2.0));
+        self.separator.size.set(size);
+        self.ellipsis.size.set(size);
+    }
+
+    fn set_default_color(&self, color: color::Rgba) {
+        self.text.set_default_color(color);
+        self.ellipsis.alpha.set(color.alpha);
+        self.separator.color.set(color.into());
+    }
+
+    fn set_font(&self, font: String) {
+        self.text.set_font(font);
+    }
+
+    fn set_default_text_size(&self, size: text::Size) {
+        self.text.set_default_text_size(size);
+    }
+
+    fn is_state_switch(&self, model: &Model) -> bool {
+        match model {
+            Model::Text(_) => self.state.get() != State::Text,
+            Model::Separator => self.state.get() != State::Separator,
+            Model::Ellipsis => self.state.get() != State::Ellipsis,
+        }
+    }
+
+    fn width(&self, text_offset: f32) -> f32 {
+        match self.state.get() {
+            State::Text => self.text.width.value() + text_offset * 2.0,
+            State::Separator => separator::ICON_WIDTH,
+            State::Ellipsis => ellipsis::ICON_WIDTH,
+        }
+    }
+}
+
+// === Params ===
+
+/// The parameters of Breadcrumbs' entries.
+#[allow(missing_docs)]
+#[derive(Clone, Debug, Default)]
+pub struct Params {
+    pub margin:                   f32,
+    pub hover_color:              color::Rgba,
+    pub font:                     ImString,
+    pub text_offset:              f32,
+    pub text_size:                text::Size,
+    pub selected_color:           color::Rgba,
+    pub highlight_corners_radius: f32,
+    pub greyed_out_color:         color::Rgba,
+    pub greyed_out_start:         Option<usize>,
+}
+
+impl Params {
+    pub(crate) fn from_style(style: &StyleWatch) -> Self {
```

Comment:
AFAIK, the `StyleWatch` is deprecated and should be replaced with `StyleWatchFrp`.

So we should have instead a method taking `StyleWatchFrp` and FRP network, which returns the `frp::Stream<Params>`. This way the breadcrumbs will be refreshed after style change.